Content.Server/_NF/Atmos/Systems/GasDepositSystem.cs

```diff
@@ -0,0 +1,315 @@
+
+        SubscribeLocalEvent<GasDepositExtractorComponent, GasPressurePumpChangeOutputPressureMessage>(
+            OnOutputPressureChangeMessage);
+        SubscribeLocalEvent<GasDepositExtractorComponent, GasPressurePumpToggleStatusMessage>(OnToggleStatusMessage);
+
+        SubscribeLocalEvent<GasSalePointComponent, AtmosDeviceUpdateEvent>(OnSalePointUpdate);
+
+        SubscribeLocalEvent<GasSaleConsoleComponent, BoundUIOpenedEvent>(OnConsoleUiOpened);
+        SubscribeLocalEvent<GasSaleConsoleComponent, GasSaleSellMessage>(OnConsoleSell);
+        SubscribeLocalEvent<GasSaleConsoleComponent, GasSaleRefreshMessage>(OnConsoleRefresh);
+    }
+
+    private void OnExtractorMapInit(Entity<GasDepositExtractorComponent> ent, ref MapInitEvent args)
+    {
+        UpdateAppearance(ent);
+    }
+
+    private void OnExtractorUiOpened(Entity<GasDepositExtractorComponent> ent, ref BoundUIOpenedEvent args)
+    {
+        Dirty(ent);
+    }
+
+    private void OnPowerChanged(Entity<GasDepositExtractorComponent> ent, ref PowerChangedEvent args)
+    {
+        UpdateAppearance(ent);
+    }
+
+    public void OnAnchorChanged(Entity<GasDepositExtractorComponent> ent, ref AnchorStateChangedEvent args)
+    {
+        if (!args.Anchored)
+            ent.Comp.DepositEntity = null;
+    }
+
+    public void OnDepositMapInit(Entity<RandomGasDepositComponent> ent, ref MapInitEvent args)
+    {
+        if (!_prototype.TryIndex(ent.Comp.DepositPrototype, out var depositPrototype))
+        {
+            if (!_prototype.TryGetRandom<GasDepositPrototype>(_random, out var randomPrototype))
+                return;
+            depositPrototype = (GasDepositPrototype)randomPrototype;
+        }
+
+        for (var i = 0; i < depositPrototype.Gases.Length && i < Atmospherics.TotalNumberOfGases; i++)
+        {
+            var gasRange = depositPrototype.Gases[i];
+            ent.Comp.Deposit.SetMoles(i, gasRange[0] + _random.NextFloat() * (gasRange[1] - gasRange[0]));
+        }
+
+        ent.Comp.LowMoles = ent.Comp.Deposit.TotalMoles * LowMoleCoefficient;
+    }
+
+    private void OnExtractorUpdate(Entity<GasDepositExtractorComponent> ent, ref AtmosDeviceUpdateEvent args)
+    {
+        if (!ent.Comp.Enabled
+            || !TryComp(ent.Comp.DepositEntity, out RandomGasDepositComponent? depositComp)
+            || TryComp<ApcPowerReceiverComponent>(ent, out var power) && !power.Powered
+            || !_nodeContainer.TryGetNode(ent.Owner, ent.Comp.PortName, out PipeNode? port)
+            || port.NodeGroup is not PipeNet { NodeCount: > 1 } net)
+        {
+            _ambientSound.SetAmbience(ent, false);
+            SetDepositState(ent, GasDepositExtractorState.Off);
+            return;
+        }
+
+        if (depositComp.Deposit.TotalMoles < Atmospherics.GasMinMoles)
+        {
+            _ambientSound.SetAmbience(ent, false);
+            SetDepositState(ent, GasDepositExtractorState.Empty);
+            return;
+        }
```

Comment:
```suggestion
            || !_nodeContainer.TryGetNode(ent.Owner, ent.Comp.PortName, out PipeNode? port))
        {
            _ambientSound.SetAmbience(ent, false);
            SetDepositState(ent, GasDepositExtractorState.Off);
            return;
        }

        if (depositComp.Deposit.TotalMoles < Atmospherics.GasMinMoles)
        {
            _ambientSound.SetAmbience(ent, false);
            SetDepositState(ent, GasDepositExtractorState.Empty);
            return;
        }
        
        if (port.NodeGroup is not PipeNet { NodeCount: > 1 } net)
        {
            _ambientSound.SetAmbience(ent, false);
            SetDepositState(ent, GasDepositExtractorState.Blocked);
            return;
        }
```
This would make the extractor glow orange (it's not pumping, but the deposit is not empty) in cases where no pipe network is attached.

Minor consistency improvement - the drill is powered but cannot run, there's nowhere for the gas to go.